7.0/rootfs/etc/php/7.0/fpm/conf.d/99-symfony.ini

```diff
@@ -0,0 +1,16 @@
+; maximum memory that OPcache can use to store compiled PHP files
+opcache.memory_consumption=256
+
+; maximum number of files that can be stored in the cache
+opcache.max_accelerated_files=20000
+
+; In production servers, PHP files should never change, unless a new application version is deployed.
+; However, by default OPcache checks if cached files have changed their contents since they were cached.
+; This check introduces some overhead that can be avoided as follows:
+opcache.validate_timestamps=0
```

Comment:
I believe we should not put this value.

I didn't see any performance gap on one of my project and it can have big side effects.

* If someone need to change a php file  urgently, the modification won't be applied.
* if someone uses php files as cache file (I know, it is not recommanded), it won't work anymore.

Reply:
> I didn't see any performance gap on one of my project and it can have big side effects.

Could you provide us some example ? Blackfire profiles ?

Reply:
I don't have a very scientific example, I just checked the response time on a login page.

Changing opcache.memory_consumption, opcache.max_accelerated_files and realpath_cache_size makes the page respond from 0.4s to 0.2s (approx).

The config opcache.validate_timestamps is unvisible on the graph.

My point is that in some (very rare) case, we have to fix a php file directly in production. Generally it is for an urgency (with a high stress level). If the change is not applied, we won't remember this configuration. We will look for a typo or something else. I believe this conf can be dangerous.

It can also introduce a BC break for some project that could implement a way to autoupdate for the code (kind of wordpress update for a Symfony project).

Reply:
I think these edge cases are too edgy to be taken into account.
This being said, maybe we could create an image variant, for example `kibatic/symfony:7.2-stateful` using a Dockerfile with only a change of value for the `opcache.validate_timestamps` property.
Or to keep the maintenance of this image easier, simply suggest in the README the line they should add to their Dockerfile if they want to be able to edit a php file in production inside their container without having to worry about calling opcache_reset() or restarting fpm ?

Reply:
> I think these edge cases are too edgy to be taken into account.

I'm pretty agree with that.

> My point is that in some (very rare) case, we have to fix a php file directly in production. Generally it is for an urgency (with a high stress level). If the change is not applied, we won't remember this configuration. We will look for a typo or something else. I believe this conf can be dangerous.

Its a common production behaviour to cache bytecode, this cannot leads us to make the prod configuration less "prod compliant".
This argument cannot be considered when choosing to setting this parameter by default. 

> For example, I'm not sure that this configuration is compatible with the PhpFileCache provided by symfony 4.1 : https://symfony.com/doc/current/components/cache.html

Stored cached data files are not php code, it's not the fact of the matter here.

> It can also introduce a BC break for some project that could implement a way to autoupdate for the code (kind of wordpress update for a Symfony project).

However this could be a real problem but I think the default should still be `0`.
You could always override this behaviour by mounting a `999-override.ini`

Maybe we should add a little bit of documentation about it anyway on how to override default php config.

WDYT @jcrombez @philippe-levan ?

Reply:
> Stored cached data files are not php code, it's not the fact of the matter here.

Hmm maybe it could be an issue.

https://github.com/symfony/cache/blob/master/Traits/PhpFilesTrait.php#L191

Btw I don't think symfony will recommend setting this value if is break cache adapter ...

We need to confirm this point

Reply:
https://github.com/symfony/symfony/blob/4.1/src/Symfony/Component/Cache/Traits/PhpFilesTrait.php#L54

Reply:
I still don't believe it is a good idea to enable it by default, but all the points are clear.
I let you choose the solution.
(but we need the readme modifications before merging)

Reply:
I'm still in favor of disabling it by default though I understand (but don't share) @philippe-levanfear fears with this choice. Maybe it would be worth to do a real bench on what good it brings to be sure it's worth the hypotethical issues of these ugly edgy edge cases before deciding something.
If it really turns out we can't measure real gain, then i'm okay with making this a enabling line in my own Dockerfile.

Reply:
As it could cause some issues, we need to to dig further more if setting this default is benefic, with maybe additional bench.

I'll remove it from this PR.

Reply:
8739e43